updates Developer Guide in README.md (#1772)
geostyler/geostyler: README.md

cd geostyler-demo
npm i
npm link geostyler
```

> When working with npm link it may happen that some tools like webpack or typescript don't know how to resolve packages that are used in booth packages (in this case `geostyler` and `geostyler-style`). So we have to configure this in the `geostyler-demo`:

Reviewer: ```suggestion
When working with npm link it may happen that some tools like webpack or typescript don't know how to resolve packages that are used in both packages (in this case `geostyler` and `geostyler-style`). So we have to configure this in the `geostyler-demo`:
```

Replace `rootDird` with `rootDirs` and add the linked packages to the `rootDirs` in `tsconfig.json`. If you link some parsers too just add them aswell.

```diff
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -9,9 +9,12 @@
     "allowJs": true,
     "jsx": "react",
     "moduleResolution": "node",
-    "rootDir": ".",
     "rootDirs": [
-      "./src/"
+      "./src/",
+      "../geostyler",
     ],
     "forceConsistentCasingInFileNames": true,
     "noImplicitReturns": true,
```

> Then uncomment or add resolve aliases to the wepackconfig:

Reviewer: ```suggestion
Then uncomment or add resolve aliases to the webpack config:
```

### Troubleshooting
`Invalid hook call` error:

> If the demo does not start but shows the above error it means that `geostyler-demo` and `geostyler` are using different react sources. Please make sure to have the react alias in the `webpack.common.config.js` configured correctly.

Reviewer: ```suggestion
If the demo does not start but shows the above error, it means that `geostyler-demo` and `geostyler` are using different react sources. Please make sure to have the react alias in the `webpack.common.config.js` configured correctly.
```
